api/v1/collector_types.go

```diff
@@ -0,0 +1,64 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// CollectorSpec defines the desired state of Collector
+type CollectorSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Type defines the type of collector that this is. Supported values are ecr_event
+	CollectorType string `json:"type"`
+	// QueueName defines the name of the queue to collect events from, in the case of the ecr_event type.
+	QueueName string `json:"queueName"`
+}
+
+// CollectorStatus defines the observed state of Collector
+type CollectorStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
```

Comment:
I could see some attribute here that could let us know whether the collector is working or not.  if its misconfigured I'd like to be able to reference something here to find out.

extra note: this is particularly useful since CRDs can be configured to output status variables alongside what is normally returned via `kubectl get`.  example: `kubectl get certificate -A` shows all certs and whether or not they're valid.

Reply:
Pushed update including this under a field 'active' I'm not sure if a better way to go about this would be to use conditions? Considered it but didn't come up with a structure I liked.

Would be happy to hear other's thoughts.